components/Button/index.jsx

```diff
@@ -32,101 +28,133 @@ export default class Button extends Component {
     isHovering: false
   }
 
-  onClickHandler = e => {
+  onClick = e => {
     e.preventDefault();
 
     if (this.props.disabled) return;
 
     this.props.onClick();
   }
 
-  onMouseOverHandler = () => {
-    if (this.props.disabled) return;
+  onMouseOver = () => {
+    if (this.props.disabled || this.props.type !== 'custom') return;
```

Comment:
@clopes-ventureoak One last thing. We can move all these if (this.props.disabled || this.props.type !== 'custom') return; into a function :)